src/main/scala/org/apache/mesos/chronos/scheduler/jobs/JobUtils.scala

```diff
@@ -104,69 +103,26 @@ object JobUtils {
     }
   }
 
-  def makeScheduleStream(job: ScheduleBasedJob, dateTime: DateTime) = {
+  /**
+   * Given a schedule stream, return a new ScheduleStream in which the head of the stream
+   * is a job which is adjusted for the given start date. That is, the recurrences in its
+   * schedule will be reduced according to the number of skips between the current start date
+   * and that provided by the datetime parameter. Of course, in the case that the new datetime
+   * is sufficiently far in the future for the job to have no more runs left, the schedulestream
+   * will have no job at its head.
+   */
+  def makeScheduleStreamForDate(job: ScheduleBasedJob, dateTime: DateTime): Option[ScheduleStream] = {
     job.schedule match {
       case schedule: ISO8601Schedule => {
         if(schedule.start.plus(job.epsilon).isBefore(dateTime)) {
-          skipForward(job, dateTime)
+          AdjustedForStartDate(schedule, dateTime).map(schedule => new ScheduleStream(schedule, job.name, job.scheduleTimeZone))
```

Comment:
Nice. The map here is kinda confusing at first, but it makes sense if you stare at it for a while :)